src/Sentry/Protocol/EnvelopeItem.cs

```diff
@@ -0,0 +1,30 @@
+using System.Text;
+
+namespace Sentry.Protocol
+{
+    public class EnvelopeItem : ISerializable
+    {
+        public EnvelopeHeaderCollection Headers { get; }
+
+        public EnvelopePayload Payload { get; }
+
+        public EnvelopeItem(EnvelopeHeaderCollection headers, EnvelopePayload payload)
+        {
+            Headers = headers;
+            Payload = payload;
+        }
+
+        public string Serialize()
+        {
+            var buffer = new StringBuilder();
```

Comment:
We could take a string builder as a parameter then and use a single builder for the whole envelope but we need to consider when we serialize blobs that are not string liek attachments